### ISSUE TITLE
bug(@angular/google-maps): caught (in promise): TypeError: Cannot read properties of undefined (reading 'remove') T

### ISSUE THREAD

**Issue by roma2341** [angular/components#26984]
### Is this a regression?

- [ ] I don't know

### Description

This critical bug causes the application to break completely in my case. If my Google Maps domain is not added to the allowed list, Google Maps does not load (which is expected), but when the "google-map" component is being destroyed, it throws an exception, causing my app to display a blank page due to the "google-map" error:
Uncaught (in promise): TypeError: Cannot read properties of undefined (reading 'remove')
```
class MapEventManager {
    constructor(_ngZone) {
        this._ngZone = _ngZone;
        /** Pending listeners that were added before the target was set. */
        this._pending = [];
        this._listeners = [];
        this._targetStream = new BehaviorSubject(undefined);
    }
    /** Clears all currently-registered event listeners. */
    _clearListeners() {
        for (const listener of this._listeners) {
            listener.remove(); //<----- NPE Error throwed from here because listener is undefined
        }
        this._listeners = [];
    }
```
A null-check should be added for the `listener.remove()` method call to avoid this error.

### Reproduction

Steps to reproduce:
1. Configure the Google Maps allowed domains list to exclude the domain where the application is hosted.
2. Open a view where Google Maps is used (the component should break here due to the disallowed domain).
3. Attempt to navigate to another view (Google Maps will throw an error on destroy, breaking further behavior, such as displaying an empty page instead of the proper view).

### Expected Behavior

The component should be destroyed without throwing any exceptions.

### Actual Behavior

When the Google Maps allowed domains list does not include the application domain, the component breaks the application where it is used.

### Environment

- Angular: 14.0.3
- @angular/google-maps: 14.2.7,
- CDK: 14.0.3
- Material: 14.2.3
- Browser(s): Chrome 112.0.5615.138 64bit
- Operating System (e.g. Windows, macOS, Ubuntu): Windows

**Comment by crisbeto** [comment]
This should be fixed in version 15.1.3 of the Google Maps package. See https://github.com/angular/components/blob/main/CHANGELOG.md#google-maps-1.

**Comment by roma2341** [comment]
As i understand Google Maps 15.1.3 doesn't support Angular 14  and we have to migrate our app to latest angular version ?
